### PR TITLE
Enable http client connection failure tests

### PR DESCRIPTION
**Pull request by laurit**
Enable http client connection failure tests for netty based http clients.
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/2367

### CODE REVIEW

File: instrumentation/netty/netty-3.8/javaagent/netty-3.8-javaagent.gradle [hunk 1]
```diff
@@ -15,18 +15,14 @@ muzzle {
   }
 }
 
-testSets {
-  latestDepTest
-}
```
Comment (trask) [inline instrumentation/netty/netty-3.8/javaagent/netty-3.8-javaagent.gradle:20]: 🎉

File: testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy [hunk 2]
```diff
@@ -813,60 +827,62 @@ abstract class HttpClientTest<REQUEST> extends InstrumentationSpecification {
       } else {
         childOf((SpanData) parentSpan)
       }
-      name expectedOperationName(method)
+      name expectedClientSpanName(uri, method)
       kind CLIENT
       if (exception) {
         status ERROR
-        errorEvent(exception.class, exception.message)
+        clientSpanErrorEvent(it, uri, exception)
```
Comment (trask) [inline testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy:834]: suggestion to rename:
```suggestion
        assertClientSpanErrorEvent(it, uri, exception)
```

File: instrumentation/netty/netty-4.1/javaagent/src/test/groovy/Netty41ClientTest.groovy [hunk 3]
```diff
@@ -173,47 +183,6 @@ class Netty41ClientTest extends HttpClientTest<DefaultFullHttpRequest> implement
     group.shutdownGracefully()
   }
 
-  // This is almost identical to "connection error (unopened port)" test from superclass.
-  // But it uses somewhat different span name for the client span.
-  // For now creating a separate test for this, hoping to remove this duplication in the future.
-  def "netty connection error (unopened port)"() {
```
Comment (trask) [inline instrumentation/netty/netty-4.1/javaagent/src/test/groovy/Netty41ClientTest.groovy:179]: 👍

**Review by trask — APPROVED** [review]
😅 👍

File: instrumentation/async-http-client/async-http-client-1.9/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/asynchttpclient/ResponseInstrumentation.java [hunk 4]
```diff
@@ -32,10 +32,15 @@
...
-    return singletonMap(
+    Map<ElementMatcher<MethodDescription>, String> transformers = new HashMap<>();
+    transformers.put(
         named("onCompleted")
             .and(takesArgument(0, named("com.ning.http.client.Response")))
             .and(isPublic()),
         ResponseAdvice.class.getName());
+    transformers.put(
+        named("onThrowable").and(takesArgument(0, Throwable.class)).and(isPublic()),
+        ResponseFailureAdvice.class.getName());
```
Comment (mateuszrzeszutek) [inline instrumentation/async-http-client/async-http-client-1.9/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/asynchttpclient/ResponseInstrumentation.java:43]:
```suggestion
        this.getClass().getPackage().getName() + ".ResponseAdvice");
    transformers.put(
        named("onThrowable").and(takesArgument(0, Throwable.class)).and(isPublic()),
        this.getClass().getPackage().getName() + ".ResponseFailureAdvice");
```

File: instrumentation/async-http-client/async-http-client-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/asynchttpclient/ResponseInstrumentation.java [hunk 5]
```diff
@@ -32,10 +32,15 @@
...
-    return singletonMap(
+    Map<ElementMatcher<MethodDescription>, String> transformers = new HashMap<>();
+    transformers.put(
         named("onCompleted")
             .and(takesArgument(0, named("org.asynchttpclient.Response")))
             .and(isPublic()),
         ResponseAdvice.class.getName());
+    transformers.put(
+        named("onThrowable").and(takesArgument(0, Throwable.class)).and(isPublic()),
+        ResponseFailureAdvice.class.getName());
```
Comment (mateuszrzeszutek) [inline instrumentation/async-http-client/async-http-client-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/asynchttpclient/ResponseInstrumentation.java:43]:
```suggestion
        this.getClass().getPackage().getName() + ".ResponseAdvice");
    transformers.put(
        named("onThrowable").and(takesArgument(0, Throwable.class)).and(isPublic()),
        this.getClass().getPackage().getName() + ".ResponseFailureAdvice");
```

File: instrumentation/vertx-web-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/client/ExceptionHandlerWrapper.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+    tracer().endExceptionally(contexts.context, throwable);
+
+    try (Scope scope = contexts.parentContext.makeCurrent()) {
+      callHandler(throwable);
+    }
+  }
+
+  private void callHandler(Throwable throwable) {
+    if (handler != null) {
```
Comment (mateuszrzeszutek) [inline instrumentation/vertx-web-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/client/ExceptionHandlerWrapper.java:45]: Can handler be null here? You only wrap it if it's not null in `ExceptionHandlerAdvice`

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**